tools/astarte_export/lib/export.ex

```diff
@@ -0,0 +1,283 @@
+  end
+
+  defp get_attributes(:item, value) do
+    %{:name => value[:v_realpathdatavalue]}
+  end
+
+  defp get_attributes(_tag, _value) do
+    %{}
+  end
+
+###############################################################
+#
+# get_value/2 function is used to construct the Inner XML tag
+# (or) data field to be used b/w opening and closing tags
+#
+###############################################################
+
+  @spec get_value( atom(), struct() | keyword() ) ::  String.t()  
+                                                     | charlist() 
+                                                     | maybe_improper_list()  
+
+  defp get_value(:astrate, options) do
+    [seralize_xml(:devices, options)]
+  end
+
+  defp get_value(:devices, [realm: realm]) do
+     {:ok, conn} = FetchData.get_connection(realm)
+     Logger.info("Connected to database.", realm: realm)
+     devices     = FetchData.get_devices(conn)
+     Logger.info("Extracted devices information from realm", realm: realm)
+     Enum.reduce(devices, [],
+       fn device_data, acc ->
+          state = FetchData.process_device_data(conn, device_data)
+          acc ++ [seralize_xml(:device, [state: state, realm: realm])]
+       end)
+  end
+
+  defp get_value(:device, options) do
+    tag_list = [:protocol,
+                :registration,
+                :credentials,
+                :stats,
+                :interfaces]
+    Enum.reduce(tag_list, [],
+      fn tag, acc ->
+        acc ++ [seralize_xml(tag, options)]
+      end)
+  end
+
+
+  defp get_value(:interfaces, options) do
+    state = options[:state]
+    Enum.reduce(state.interfaces , [],
+      fn interface_state, acc ->
+        acc ++ [seralize_xml(:interface, interface_state)]
+      end)
+  end
+
+  defp get_value(:interface, interface_state) do
+    mappings = interface_state.mappings
+    Enum.reduce(mappings, [],
+      fn mapping, acc ->
+       output =
+        case mapping.type do
+          {:datastream, _} ->
+            seralize_xml(:datastream, mapping)
+          {:properties, _ } ->
+                Enum.reduce( mapping.value, [],
+                  fn value, acc ->
+                     acc ++[seralize_xml(:property, value)]
+                 end)
+        end
+        case mapping.type do
+           {:datastream, _} ->
+               acc  ++ [output]
+           {:properties, _} ->
+               List.flatten(acc ++ [output]) 
+        end
+      end)
+  end
+
+  defp get_value(:datastream, mapping) do
+   Enum.reduce(mapping.value, [],
+     fn value, acc ->
+       output =
+       case mapping.type do 
+         {_, :object} ->
+            seralize_xml(:object, value)
+         {_, :individual} ->
+            seralize_xml(:value, value)
+       end
+       acc ++ [output]
+    end)
+  end
+
+  defp get_value(:object, value) do
+      [seralize_xml(:item, value)]
+  end
+
+  defp get_value(:property, value) do
+     value[:double_value] |> Kernel.to_string
```

Comment:
Why `Kernel.to_string()` and not just `to_string()`?

Reply:
Both are same.. just calling with Module.function_name

Reply:
updated to to_string() itself